prepare.sh

```diff
@@ -37,7 +37,11 @@ if ! [[ $(enroot list | grep "${CONTAINER_NAME}") ]]; then
     # Import a container image from a specific location to enroot image dir
     # Scheme: docker://[USER@][REGISTRY#]IMAGE[:TAG]
     IMAGE_DIR="${ENROOT_DATA_PATH}"
-    URL="docker://${CUSTOM_ENV_CI_JOB_IMAGE}"
+    if [[ ${CUSTOM_ENV_CI_JOB_IMAGE} == "${CUSTOM_ENV_CI_REGISTRY}"* ]]; then
+        URL="docker://${CUSTOM_ENV_CI_REGISTRY_USER}:${CUSTOM_ENV_CI_REGISTRY_PASSWORD}@${CUSTOM_ENV_CI_REGISTRY}#${CUSTOM_ENV_CI_JOB_IMAGE#*$CUSTOM_ENV_CI_REGISTRY/}"
```

Comment:
That requires storing the registry password as a variable in either the `.gitlab-ci.yml` script or the `.gitlab-runner/config.toml` file, right? For safety, Is there a way to make this optional and rely on the `$ENROOT_CONFIG_PATH/.credentials` when the user wants to instead?

Reply:
No CI_REGISTRY_PASSWORD is the default token:

https://docs.gitlab.com/ee/ci/variables/